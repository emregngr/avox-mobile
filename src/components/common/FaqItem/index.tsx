import { MaterialCommunityIcons } from '@expo/vector-icons'
import { useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'
import Collapsible from 'react-native-collapsible'

import { ThemedText } from '@/components/common/ThemedText'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { FaqItemType } from '@/types/common/faq'

interface FaqItemProps {
  index: number
  isExpanded: boolean
  item: FaqItemType
  toggleExpanded: (index: number) => void
}

export const FaqItem = ({ index, isExpanded, item, toggleExpanded }: FaqItemProps) => {
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { description, title } = item
  return (
    <View className="mb-2">
      <TouchableOpacity
        activeOpacity={0.7}
        className="p-4 rounded-xl overflow-hidden bg-background-secondary"
        hitSlop={10}
        onPress={() => toggleExpanded(index)}
        testID={`faq-${item?.id}`}
      >
        <View className="flex-row justify-between items-center">
          <View className="flex-1 pr-4">
            <ThemedText color="text-100" type="body1">
              {title}
            </ThemedText>
          </View>
          <MaterialCommunityIcons
            color={colors?.onPrimary50}
            name={isExpanded ? 'chevron-up' : 'chevron-down'}
            size={24}
          />
        </View>
      </TouchableOpacity>

      <Collapsible collapsed={!isExpanded}>
        <View className="mt-2 p-4 rounded-xl overflow-hidden bg-background-tertiary">
          <ThemedText color="text-90" type="body2">
            {description}
          </ThemedText>
        </View>
      </Collapsible>
    </View>
  )
}
