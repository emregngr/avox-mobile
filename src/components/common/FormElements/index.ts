export * from './CheckboxField'
export * from './TextInputField'
