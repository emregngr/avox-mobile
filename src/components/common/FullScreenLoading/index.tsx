import React from 'react'
import { ActivityIndicator, View } from 'react-native'

export const FullScreenLoading = () => (
  <View
    className="flex-1 items-center justify-center bg-background-primary"
    testID="full-screen-loading"
  >
    <ActivityIndicator className="color-text-100" size="large" testID="loading-indicator" />
  </View>
)
