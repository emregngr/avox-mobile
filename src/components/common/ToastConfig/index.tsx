import { useMemo } from 'react'
import type { ToastProps } from 'react-native-toast-message'
import { ErrorToast, SuccessToast } from 'react-native-toast-message'

import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'

interface ToastConfigProps extends ToastProps {
  style?: any
  text1Props?: any
  text1Style?: any
  text2NumberOfLines?: number
  text2Props?: any
  text2Style?: any
}

export const useToastConfig = () => {
  const { selectedTheme } = useThemeStore()
  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const toastConfig = useMemo(() => {
    const baseToastProps = {
      text1Props: { allowFontScaling: false, maxFontSizeMultiplier: 1 },
      text2NumberOfLines: 2,
      text2Props: { allowFontScaling: false, maxFontSizeMultiplier: 1 },
    }

    const baseStyle = {
      backgroundColor: colors?.background?.tertiary,
      height: 70,
      marginTop: 20,
    }

    const baseTextStyle = {
      text1Style: {
        color: colors?.text100,
        fontFamily: 'Inter-Bold',
        fontSize: 16,
        lineHeight: 18,
      },
      text2Style: {
        color: colors?.text100,
        fontFamily: 'Inter-Regular',
        fontSize: 14,
        lineHeight: 16,
      },
    }

    return {
      error: (props: ToastConfigProps) => (
        <ErrorToast
          {...props}
          {...baseToastProps}
          {...baseTextStyle}
          style={{
            ...baseStyle,
            borderLeftColor: colors?.error,
          }}
        />
      ),
      success: (props: ToastConfigProps) => (
        <SuccessToast
          {...props}
          {...baseToastProps}
          {...baseTextStyle}
          style={{
            ...baseStyle,
            borderLeftColor: colors?.success,
          }}
        />
      ),
    }
  }, [colors])

  return toastConfig
}
