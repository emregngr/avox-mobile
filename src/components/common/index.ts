export * from './FaqItem'
export * from './FormElements'
export * from './FullScreenLoading'
export * from './Header'
export * from './OnboardingItem'
export * from './ProfileItem'
export * from './RenderDetailTabBar'
export * from './RenderTabBar'
export * from './SafeLayout'
export * from './SearchInput'
export * from './ThemedButton'
export * from './ThemedButtonText'
export * from './ThemedGradientButton'
export * from './ThemedText'
export * from './ToastConfig'
