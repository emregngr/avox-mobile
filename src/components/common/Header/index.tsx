import { GlassView } from 'expo-glass-effect'
import * as Haptics from 'expo-haptics'
import { type ReactNode, useMemo } from 'react'
import type { ViewStyle } from 'react-native'
import { TouchableOpacity, View } from 'react-native'

import Back from '@/assets/icons/back.svg'
import { ThemedButtonText } from '@/components/common/ThemedButtonText'
import { ThemedText } from '@/components/common/ThemedText'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import { cn } from '@/utils/common/cn'

type HeaderProps = {
  backIcon?: boolean
  backIconOnPress?: () => void
  containerClassName?: string
  hapticFeedback?: boolean
  isFavorite?: boolean
  rightButtonLabel?: string
  rightButtonOnPress?: () => void
  rightIcon?: ReactNode
  rightIconOnPress?: () => void
  shareIcon?: ReactNode
  shareIconOnPress?: () => void
  style?: ViewStyle
  testID?: string
  title?: string | string[]
  titleClassName?: string
}

export const Header = ({
  backIcon = true,
  backIconOnPress,
  containerClassName,
  hapticFeedback = false,
  isFavorite = false,
  rightButtonLabel,
  rightButtonOnPress,
  rightIcon,
  rightIconOnPress,
  shareIcon,
  shareIconOnPress,
  style,
  testID,
  title,
  titleClassName,
}: HeaderProps) => {
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const STATIC_STYLES = {
    backIcon: {
      left: 16,
      position: 'absolute' as const,
      borderRadius: 20,
      padding: 8,
    },
    shareIcon: {
      right: 64,
      position: 'absolute' as const,
      borderRadius: 20,
      padding: 8,
    },
    rightIcon: {
      right: 16,
      position: 'absolute' as const,
      borderRadius: 20,
      padding: 8,
    },
  }

  const handleRightIconOnPress = () => {
    if (hapticFeedback) {
      if (isFavorite) {
        Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light)
      } else {
        Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Medium)
      }
    }

    rightIconOnPress?.()
  }

  return (
    <View
      className={cn('h-11 justify-center items-center', containerClassName)}
      style={style}
      testID={testID || 'header'}
    >
      {title ? (
        <ThemedText
          className={titleClassName ?? ''}
          color="text-100"
          ellipsizeMode="tail"
          numberOfLines={2}
          type="h3"
          center
        >
          {title}
        </ThemedText>
      ) : null}

      {backIcon ? (
        <GlassView
          style={STATIC_STYLES.backIcon}
          tintColor={colors?.background?.glass}
          glassEffectStyle="clear"
          isInteractive
        >
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={20}
            onPress={backIconOnPress}
            testID="header-back-icon"
          >
            <Back color={colors?.onPrimary100} height={24} width={24} />
          </TouchableOpacity>
        </GlassView>
      ) : null}

      {shareIcon ? (
        <GlassView
          style={STATIC_STYLES.shareIcon}
          tintColor={colors?.background?.glass}
          glassEffectStyle="clear"
          isInteractive
        >
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={shareIconOnPress}
            testID="header-share-icon"
          >
            {shareIcon}
          </TouchableOpacity>
        </GlassView>
      ) : null}

      {rightButtonLabel ? (
        <ThemedButtonText
          containerStyle="absolute right-4"
          label={rightButtonLabel}
          onPress={rightButtonOnPress as () => void}
          textColor="text-100"
          type="h4"
        />
      ) : rightIcon ? (
        <GlassView
          style={STATIC_STYLES.rightIcon}
          tintColor={colors?.background?.glass}
          glassEffectStyle="clear"
          isInteractive
        >
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={handleRightIconOnPress}
            testID="header-right-icon"
          >
            {rightIcon}
          </TouchableOpacity>
        </GlassView>
      ) : null}
    </View>
  )
}
