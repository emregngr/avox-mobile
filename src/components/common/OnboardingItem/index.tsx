import { Image } from 'expo-image'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import type { OnBoardingType } from '@/types/common/onBoarding'

interface OnboardingItemProps {
  item: OnBoardingType
}

const STATIC_STYLES = {
  image: {
    height: '45%' as const,
    marginVertical: 20,
    width: '100%' as const,
  },
}

export const OnboardingItem = ({ item }: OnboardingItemProps) => {
  const { image, text, title } = item
  return (
    <View className="w-screen h-screen bg-background-primary">
      <Image
        cachePolicy="memory-disk"
        contentFit="contain"
        source={image}
        style={STATIC_STYLES.image}
        transition={0}
      />
      <View className="mx-4 z-10">
        <ThemedText
          className="mb-4"
          color="text-100"
          lineBreakMode="tail"
          numberOfLines={2}
          type="h1"
        >
          {title}
        </ThemedText>
        <ThemedText
          color="text-100" lineBreakMode="tail" numberOfLines={4}
          type="h3"
        >
          {text}
        </ThemedText>
      </View>
    </View>
  )
}
