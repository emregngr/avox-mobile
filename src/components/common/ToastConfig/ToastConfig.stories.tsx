import type { Meta, StoryObj } from '@storybook/react-native'
import React from 'react'
import { ScrollView, View } from 'react-native'
import Toast from 'react-native-toast-message'

import { ThemedGradientButton } from '@/components/common/ThemedGradientButton'
import { ThemedText } from '@/components/common/ThemedText'
import { useToastConfig } from '@/components/common/ToastConfig'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'

const ToastConfigDemo = () => {
  const { selectedTheme } = useThemeStore()
  const colors = themeColors?.[selectedTheme]

  const toastConfig = useToastConfig()

  const showSuccessToast = (title: string, message: string) => {
    Toast.show({
      type: 'success',
      text1: title,
      text2: message,
    })
  }

  const showErrorToast = (title: string, message: string) => {
    Toast.show({
      type: 'error',
      text1: title,
      text2: message,
    })
  }

  return (
    <View className="flex-1 p-4">
      <ScrollView contentContainerClassName="gap-4" showsVerticalScrollIndicator={false}>
        <View>
          <ThemedText type="h2" color="text-100" className="mb-4">
            Toast Configuration Demo
          </ThemedText>
          <ThemedText type="body2" color="text-70" className="mb-6">
            Test different toast types and configurations. The toasts will appear at the top of the
            screen.
          </ThemedText>
        </View>

        <View>
          <ThemedText type="h3" color="text-100" className="mb-3">
            Success Toasts
          </ThemedText>

          <View className="gap-3">
            <ThemedGradientButton
              label="Show Success Toast"
              type="custom"
              customGradientColors={[colors?.success, colors?.success]}
              onPress={() => showSuccessToast('Success!', 'Operation completed successfully')}
            />

            <ThemedGradientButton
              label="Long Success Message"
              type="custom"
              customGradientColors={[colors?.success, colors?.success]}
              onPress={() =>
                showSuccessToast(
                  'Data Sync Complete',
                  'Your data has been synchronized across all devices and is now up to date',
                )
              }
            />

            <ThemedGradientButton
              label="Simple Success"
              type="custom"
              customGradientColors={[colors?.success, colors?.success]}
              onPress={() => showSuccessToast('Saved', 'Changes saved')}
            />
          </View>
        </View>

        <View>
          <ThemedText type="h3" color="text-100" className="mb-3">
            Error Toasts
          </ThemedText>

          <View className="gap-3">
            <ThemedGradientButton
              label="Show Error Toast"
              type="custom"
              customGradientColors={[colors?.error, colors?.error]}
              onPress={() => showErrorToast('Error!', 'Something went wrong')}
            />

            <ThemedGradientButton
              label="Network Error"
              type="custom"
              customGradientColors={[colors?.error, colors?.error]}
              onPress={() =>
                showErrorToast(
                  'Connection Failed',
                  'Unable to connect to server. Please check your internet connection and try again',
                )
              }
            />

            <ThemedGradientButton
              label="Validation Error"
              type="custom"
              customGradientColors={[colors?.error, colors?.error]}
              onPress={() => showErrorToast('Invalid Input', 'Please fill in all required fields')}
            />
          </View>
        </View>

        <View>
          <ThemedText type="h3" color="text-100" className="mb-3">
            Real-world Examples
          </ThemedText>

          <View className="gap-3">
            <ThemedGradientButton
              label="Login Success"
              type="primary"
              onPress={() =>
                showSuccessToast('Welcome Back!', 'You have been successfully logged in')
              }
            />

            <ThemedGradientButton
              label="File Upload"
              type="primary"
              onPress={() =>
                showSuccessToast('Upload Complete', 'Your file has been uploaded and processed')
              }
            />

            <ThemedGradientButton
              label="Payment Error"
              type="custom"
              customGradientColors={[colors?.error, colors?.error]}
              onPress={() =>
                showErrorToast(
                  'Payment Failed',
                  'Your card was declined. Please try a different payment method',
                )
              }
            />

            <ThemedGradientButton
              label="Server Error"
              type="custom"
              customGradientColors={[colors?.error, colors?.error]}
              onPress={() =>
                showErrorToast(
                  'Server Error (500)',
                  'Internal server error. Our team has been notified',
                )
              }
            />
          </View>
        </View>

        <View>
          <ThemedText type="h3" color="text-100" className="mb-3">
            Toast Timing
          </ThemedText>

          <View className="gap-3">
            <ThemedGradientButton
              label="Quick Toast (1s)"
              type="secondary"
              onPress={() => {
                Toast.show({
                  type: 'success',
                  text1: 'Quick Message',
                  text2: 'This will disappear in 1 second',
                  visibilityTime: 1000,
                })
              }}
            />

            <ThemedGradientButton
              label="Long Toast (5s)"
              type="secondary"
              onPress={() => {
                Toast.show({
                  type: 'error',
                  text1: 'Important Notice',
                  text2: 'This message will stay for 5 seconds',
                  visibilityTime: 5000,
                })
              }}
            />
          </View>
        </View>

        <View>
          <ThemedText type="h3" color="text-100" className="mb-3">
            Multiple Toasts
          </ThemedText>

          <View className="gap-3">
            <ThemedGradientButton
              label="Show Sequential Toasts"
              type="custom"
              customGradientColors={[colors?.primaryGradientStart, colors?.primaryGradientEnd]}
              onPress={() => {
                showSuccessToast('Step 1', 'First operation completed')

                setTimeout(() => {
                  showSuccessToast('Step 2', 'Second operation completed')
                }, 1500)

                setTimeout(() => {
                  showSuccessToast('Complete', 'All operations finished')
                }, 3000)
              }}
            />

            <ThemedGradientButton
              label="Mixed Toast Types"
              type="custom"
              customGradientColors={[colors?.warning, colors?.warning]}
              onPress={() => {
                showSuccessToast('Success', 'First part completed')

                setTimeout(() => {
                  showErrorToast('Warning', 'Second part had issues')
                }, 2000)
              }}
            />
          </View>
        </View>

        <View className="h-[100px]" />
      </ScrollView>

      <Toast config={toastConfig} />
    </View>
  )
}

const meta: Meta<typeof ToastConfigDemo> = {
  title: 'Components/ToastConfig',
  component: ToastConfigDemo,
  parameters: {
    docs: {
      description: {
        component:
          'A hook that provides themed toast configuration for success and error messages with consistent styling.',
      },
    },
  },
  decorators: [
    Story => {
      return (
        <View className="flex-1 bg-background-primary">
          <Story />
        </View>
      )
    },
  ],
}

export default meta

type Story = StoryObj<typeof ToastConfigDemo>

export const BasicUsage: Story = {
  render: () => <ToastConfigDemo />,
}

export const SuccessToasts: Story = {
  render: () => {
    const { selectedTheme } = useThemeStore()
    const colors = themeColors?.[selectedTheme]

    const toastConfig = useToastConfig()

    const examples = [
      { title: 'Simple Success', message: 'Task completed' },
      { title: 'Data Saved', message: 'Your changes have been saved successfully' },
      { title: 'Upload Complete', message: 'File uploaded and processed successfully' },
      {
        title: 'Settings Updated',
        message: 'Your preferences have been updated and will take effect immediately',
      },
    ]

    return (
      <View className="flex-1 p-4">
        <ThemedText type="h2" color="text-100" className="mb-4">
          Success Toast Examples
        </ThemedText>

        <ScrollView contentContainerClassName="gap-4" showsVerticalScrollIndicator={false}>
          {examples.map((example, index) => (
            <ThemedGradientButton
              key={index}
              label={example.title}
              type="custom"
              customGradientColors={[colors?.success, colors?.success]}
              onPress={() => {
                Toast.show({
                  type: 'success',
                  text1: example.title,
                  text2: example.message,
                })
              }}
            />
          ))}
        </ScrollView>

        <Toast config={toastConfig} />
      </View>
    )
  },
}

export const ErrorToasts: Story = {
  render: () => {
    const { selectedTheme } = useThemeStore()
    const colors = themeColors?.[selectedTheme]

    const toastConfig = useToastConfig()

    const examples = [
      { title: 'Network Error', message: 'Unable to connect to server' },
      { title: 'Validation Error', message: 'Please check your input and try again' },
      { title: 'Permission Denied', message: 'You do not have permission to perform this action' },
      { title: 'Server Timeout', message: 'Request timed out. Please try again later' },
    ]

    return (
      <View className="flex-1 p-4">
        <ThemedText type="h2" color="text-100" className="mb-4">
          Error Toast Examples
        </ThemedText>

        <ScrollView contentContainerClassName="gap-4" showsVerticalScrollIndicator={false}>
          {examples.map((example, index) => (
            <ThemedGradientButton
              key={index}
              label={example.title}
              type="custom"
              customGradientColors={[colors?.error, colors?.error]}
              onPress={() => {
                Toast.show({
                  type: 'error',
                  text1: example.title,
                  text2: example.message,
                })
              }}
            />
          ))}
        </ScrollView>

        <Toast config={toastConfig} />
      </View>
    )
  },
}

export const ConfigurationShowcase: Story = {
  render: () => {
    const { selectedTheme } = useThemeStore()
    const colors = themeColors?.[selectedTheme]

    const toastConfig = useToastConfig()

    return (
      <View className="flex-1 p-4">
        <ThemedText type="h2" color="text-100" className="mb-2">
          Toast Configuration
        </ThemedText>
        <ThemedText type="body2" color="text-70" className="mb-6">
          Demonstrates the themed styling and configuration options
        </ThemedText>

        <ScrollView contentContainerClassName="gap-4" showsVerticalScrollIndicator={false}>
          <View>
            <ThemedText type="h4" color="text-100" className="mb-2">
              Standard Configuration
            </ThemedText>
            <ThemedText type="body3" color="text-70" className="mb-3">
              • 70px height, 12px border radius • Theme-based background and text colors • Font
              scaling disabled for consistency • 2-line text support
            </ThemedText>
            <ThemedGradientButton
              label="Show Standard Toast"
              type="primary"
              onPress={() => {
                Toast.show({
                  type: 'success',
                  text1: 'Standard Configuration',
                  text2: 'This uses the default toast styling with theme colors',
                })
              }}
            />
          </View>

          <View>
            <ThemedText type="h4" color="text-100" className="mb-2">
              Long Text Handling
            </ThemedText>
            <ThemedText type="body3" color="text-70" className="mb-3">
              Text automatically wraps to 2 lines maximum
            </ThemedText>
            <ThemedGradientButton
              label="Show Long Text"
              type="secondary"
              onPress={() => {
                Toast.show({
                  type: 'error',
                  text1: 'Very Long Title That Might Wrap',
                  text2:
                    'This is a very long message that demonstrates how the toast handles text that exceeds the normal length and might need to wrap to multiple lines',
                })
              }}
            />
          </View>

          <View>
            <ThemedText type="h4" color="text-100" className="mb-2">
              Custom Positioning
            </ThemedText>
            <ThemedText type="body3" color="text-70" className="mb-3">
              20px top margin with device-width responsive sizing
            </ThemedText>
            <ThemedGradientButton
              label="Show Positioned Toast"
              type="custom"
              customGradientColors={[colors?.primaryGradientStart, colors?.secondaryGradientEnd]}
              onPress={() => {
                Toast.show({
                  type: 'success',
                  text1: 'Positioned Toast',
                  text2: 'Notice the consistent positioning and sizing',
                  position: 'top',
                })
              }}
            />
          </View>
        </ScrollView>

        <Toast config={toastConfig} />
      </View>
    )
  },
}
