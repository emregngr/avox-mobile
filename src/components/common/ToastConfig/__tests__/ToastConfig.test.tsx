import { renderHook } from '@testing-library/react-hooks'
import { render } from '@testing-library/react-native'
import type { ToastProps } from 'react-native-toast-message'

import { useToastConfig } from '@/components/common/ToastConfig'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'

jest.mock('@/store/theme')

const mockedUseThemeStore = useThemeStore as jest.MockedFunction<typeof useThemeStore>

const darkColors = themeColors.dark

const lightColors = themeColors.light

jest.mock('@/utils/common/responsive', () => ({
  responsive: {
    deviceWidth: 375,
  },
}))

jest.mock('react-native-toast-message', () => {
  const { View } = require('react-native')

  return {
    ErrorToast: (props: ToastProps) => <View testID="mocked-error-toast" {...props} />,
    SuccessToast: (props: ToastProps) => <View testID="mocked-success-toast" {...props} />,
  }
})

beforeEach(() => {
  mockedUseThemeStore.mockReturnValue({ selectedTheme: 'light' })
})

describe('useToastConfig Component', () => {
  it('should return a config object with success and error render functions', () => {
    const { result } = renderHook(() => useToastConfig())

    expect(result.current).toBeInstanceOf(Object)
    expect(result.current.success).toBeInstanceOf(Function)
    expect(result.current.error).toBeInstanceOf(Function)
  })

  it('should configure SuccessToast with correct props for the light theme', () => {
    const { result } = renderHook(() => useToastConfig())

    const SuccessToastComponent = result.current.success({})
    const { getByTestId } = render(SuccessToastComponent)
    const toast = getByTestId('mocked-success-toast')

    expect(toast.props.text2NumberOfLines).toBe(2)
    expect(toast.props.text1Style.color).toBe(lightColors.text100)
    expect(toast.props.text2Style.color).toBe(lightColors.text90)

    expect(toast.props.style.backgroundColor).toBe(lightColors.background.tertiary)
    expect(toast.props.style.borderLeftColor).toBe(lightColors.success)
    expect(toast.props.style.width).toBe(375 - 32)
  })

  it('should configure ErrorToast with the correct error color', () => {
    const { result } = renderHook(() => useToastConfig())

    const ErrorToastComponent = result.current.error({})
    const { getByTestId } = render(ErrorToastComponent)
    const toast = getByTestId('mocked-error-toast')

    expect(toast.props.style.borderLeftColor).toBe(lightColors.error)
  })

  it('should pass through custom props to the underlying toast component', () => {
    const { result } = renderHook(() => useToastConfig())
    const customProps = { text1: 'Custom Title', text2: 'My custom message.' }

    const SuccessToastComponent = result.current.success(customProps as any)
    const { getByTestId } = render(SuccessToastComponent)
    const toast = getByTestId('mocked-success-toast')

    expect(toast.props.text1).toBe('Custom Title')
    expect(toast.props.text2).toBe('My custom message.')
  })

  it('should update toast styles when the theme changes from light to dark', () => {
    const { result, rerender } = renderHook(() => useToastConfig())

    mockedUseThemeStore.mockReturnValue({ selectedTheme: 'dark' })

    rerender()

    const SuccessToastComponent = result.current.success({})
    const { getByTestId } = render(SuccessToastComponent)
    const toast = getByTestId('mocked-success-toast')

    expect(toast.props.text1Style.color).toBe(darkColors.text100)
    expect(toast.props.style.backgroundColor).toBe(darkColors.background.tertiary)
    expect(toast.props.style.borderLeftColor).toBe(darkColors.success)
  })
})

describe('useToastConfig Component Snapshot', () => {
  it('should render the useToastConfig Component successfully', () => {
    const { result } = renderHook(() => useToastConfig())

    expect(result.current.success({})).toMatchSnapshot()
    expect(result.current.error({})).toMatchSnapshot()
  })
})
