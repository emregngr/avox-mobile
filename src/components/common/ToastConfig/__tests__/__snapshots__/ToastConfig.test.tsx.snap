// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`useToastConfig Component Snapshot should render the useToastConfig Component successfully 1`] = `
{
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": {},
  "key": null,
  "props": {
    "style": {
      "backgroundColor": "#E5E5EA",
      "borderLeftColor": "#86EFAC",
      "borderRadius": 12,
      "height": 70,
      "marginTop": 20,
      "width": 343,
    },
    "text1Props": {
      "allowFontScaling": false,
      "maxFontSizeMultiplier": 1,
    },
    "text1Style": {
      "color": "#000000",
      "fontFamily": "Inter-Bold",
      "fontSize": 16,
      "lineHeight": 18,
    },
    "text2NumberOfLines": 2,
    "text2Props": {
      "allowFontScaling": false,
      "maxFontSizeMultiplier": 1,
    },
    "text2Style": {
      "color": "#000000E5",
      "fontFamily": "Inter-Regular",
      "fontSize": 14,
      "lineHeight": 16,
    },
  },
  "type": [Function],
}
`;

exports[`useToastConfig Component Snapshot should render the useToastConfig Component successfully 2`] = `
{
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": {},
  "key": null,
  "props": {
    "style": {
      "backgroundColor": "#E5E5EA",
      "borderLeftColor": "#E57373",
      "borderRadius": 12,
      "height": 70,
      "marginTop": 20,
      "width": 343,
    },
    "text1Props": {
      "allowFontScaling": false,
      "maxFontSizeMultiplier": 1,
    },
    "text1Style": {
      "color": "#000000",
      "fontFamily": "Inter-Bold",
      "fontSize": 16,
      "lineHeight": 18,
    },
    "text2NumberOfLines": 2,
    "text2Props": {
      "allowFontScaling": false,
      "maxFontSizeMultiplier": 1,
    },
    "text2Style": {
      "color": "#000000E5",
      "fontFamily": "Inter-Regular",
      "fontSize": 14,
      "lineHeight": 16,
    },
  },
  "type": [Function],
}
`;
